### ISSUE TITLE
On first certificate only: Can't open /{any_path}/easyrsa3/pki/index.txt.attr for reading, No such file or directory

### ISSUE THREAD

**Issue by TinCanTech** [OpenVPN/easy-rsa#199]
When creating the first certificate in a new PKI the `index.txt.attr` file does not exist which results in the error below.  The first certificate is successfully created anyway and creating subsequent certs does **not** result in the same error.
Tested: Easyrsa 305 and 304 -- same result.
(Not sure if this has been reported before, could not find an issue for it)

```
tct@deb9:~/easyrsa/ersa-305-default$ cd easyrsa3
tct@deb9:~/easyrsa/ersa-305-default/easyrsa3$ cp vars.example vars
tct@deb9:~/easyrsa/ersa-305-default/easyrsa3$ ./easyrsa init-pki

Note: using Easy-RSA configuration from: ./vars


WARNING!!!

You are about to remove the EASYRSA_PKI at: /home/tct/easyrsa/ersa-305-default/easyrsa3/pki
and initialize a fresh PKI here.

Type the word 'yes' to continue, or any other input to abort.
  Confirm removal: yes

init-pki complete; you may now create a CA or requests.
Your newly created PKI dir is: /home/tct/easyrsa/ersa-305-default/easyrsa3/pki

tct@deb9:~/easyrsa/ersa-305-default/easyrsa3$ ./easyrsa build-ca nopass

Note: using Easy-RSA configuration from: ./vars
Generating RSA private key, 2048 bit long modulus
...............................................................+++
........................................................................................................................................................................+++
e is 65537 (0x010001)
You are about to be asked to enter information that will be incorporated
into your certificate request.
What you are about to enter is what is called a Distinguished Name or a DN.
There are quite a few fields but you can leave some blank
For some fields there will be a default value,
If you enter '.', the field will be left blank.
-----
Common Name (eg: your user, host, or server name) [Easy-RSA CA]:

CA creation complete and you may now import and sign cert requests.
Your new CA certificate file for publishing is at:
/home/tct/easyrsa/ersa-305-default/easyrsa3/pki/ca.crt

tct@deb9:~/easyrsa/ersa-305-default/easyrsa3$ ./easyrsa build-server-full v305.rsa.srv.01 nopass

Note: using Easy-RSA configuration from: ./vars
Generating a 2048 bit RSA private key
...................................+++
..................................................+++
writing new private key to '/home/tct/easyrsa/ersa-305-default/easyrsa3/pki/private/v305.rsa.srv.01.key.P3gdTF8iCt'
-----
Using configuration from ./openssl-easyrsa.cnf

*** ERROR ***

Can't open /home/tct/easyrsa/ersa-305-default/easyrsa3/pki/index.txt.attr for reading, No such file or directory
140203623146752:error:02001002:system library:fopen:No such file or directory:../crypto/bio/bss_file.c:74:fopen('/home/tct/easyrsa/ersa-305-default/easyrsa3/pki/index.txt.attr','r')
140203623146752:error:2006D080:BIO routines:BIO_new_file:no such file:../crypto/bio/bss_file.c:81:
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'v305.rsa.srv.01'
Certificate is to be certified until Mar 30 16:06:00 2028 GMT (3650 days)

Write out database with 1 new entries
Data Base Updated

*** Second server cert ***

tct@deb9:~/easyrsa/ersa-305-default/easyrsa3$ ./easyrsa build-server-full v305.rsa.srv.02 nopass

Note: using Easy-RSA configuration from: ./vars
Generating a 2048 bit RSA private key
.+++
...............................................+++
writing new private key to '/home/tct/easyrsa/ersa-305-default/easyrsa3/pki/private/v305.rsa.srv.02.key.HRc4HOa1t7'
-----
Using configuration from ./openssl-easyrsa.cnf
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'v305.rsa.srv.02'
Certificate is to be certified until Mar 30 16:14:28 2028 GMT (3650 days)

Write out database with 1 new entries
Data Base Updated
```

**Comment by ManOfFlash** [comment]
I create dummy index.txt.attr and this error was gone. On Window.

**Comment by jogc** [comment]
I got this as well. Debian 9.0, EasyRSA 3.0.6:

# ./easyrsa sign-req server xxx

Using SSL: openssl OpenSSL 1.1.0j  20 Nov 2018


You are about to sign the following certificate.
Please check over the details shown below for accuracy. Note that this request
has not been cryptographically verified. Please be sure it came from a trusted
source or that you have verified the request checksum with the sender.

Request subject, to be signed as a server certificate for 1080 days:

subject=
    commonName                = xxx


Type the word 'yes' to continue, or any other input to abort.
  Confirm request details: yes
Using configuration from /root/EasyRSA-v3.0.6/pki/safessl-easyrsa.cnf
Can't open /root/EasyRSA-v3.0.6/pki/index.txt.attr for reading, No such file or directory
140360256168000:error:02001002:system library:fopen:No such file or directory:../crypto/bio/bss_file.c:74:fopen('/root/EasyRSA-v3.0.6/pki/index.txt.attr','r')
140360256168000:error:2006D080:BIO routines:BIO_new_file:no such file:../crypto/bio/bss_file.c:81:
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'xxx'
Certificate is to be certified until May 31 11:14:19 2022 GMT (1080 days)

Write out database with 1 new entries
Data Base Updated

Certificate created at: /root/EasyRSA-v3.0.6/pki/issued/xxx.crt

#

**Comment by TinCanTech** [comment]
This issue is fixed in openssl 1.1.1d
